app/services/teacher_training_api/import_course.rb

```diff
@@ -50,7 +49,13 @@ def invalid_age_range?
     end
 
     def subjects
-      Subject.where(code: course_attributes[:subject_codes])
+      codes = course_attributes[:subject_codes]
+      # It's critical that the ordering of course.subjects ends up matching the order of course_attributes[:subject_codes]
+      # Why? Because the first subject will be mapped to an allocation subject and the allocation subject determines
+      # what bursary the trainee is entitled to. By ordering it correctly here, ActiveRecord will create the association
+      # records in the same order, and then we just need to add order("course_subjects.created_at") to Course#subjects.
+      # Adding an explicit "position" column to the course_subjects table was explored, but it's lot more complicated.
+      Subject.where(code: codes).sort_by { |subject| codes.index(subject.code) }
```

Comment:
When restoring databases, will the order be ordered as expected?

Reply:
Yes, but the timestamps should be preserved, right?

Reply:
Normally it's order it by `id`, as that's is the only guaranteed as atomic operation, as database is not guaranteed to returned in any order unless an order by clause is used.